### ISSUE TITLE
How are the `alpha` values decided for the DPR experiments?

### ISSUE THREAD

**Issue by ccsasuke** [castorini/pyserini#728]
Hi,

On this page https://github.com/castorini/pyserini/blob/master/docs/experiments-dpr.md, a different `alpha` value is used for each dataset in the hybrid model. Could you share the details on how the `alpha` values are chosen for each dataset?
Did you tune it on the validation sets? If so, how is it done in particular?

Thank you.

**Comment by MXueguang** [comment]
For NQ, TQA, WQ, Curated, we use the entire train set to tune.
For Squad, we use a subset of the train set to tune.
`alpha` is tuned in range(0,2) with line search.

**Comment by ccsasuke** [comment]
@MXueguang Thanks for the prompt response!

If I understand correctly, you select the best value by running hybrid retrieval using a range of different `alpha` values, and picked one that's "optimal", right?

What values did you consider? (0.1, 0.2, ..., 2.0 ?)
Which metric did you use to select the best value? Did you use top-100 retrieval accuracy?

**Comment by MXueguang** [comment]
We tuned the α values on different datasets by optimizing top20 retrieval accuracy: For datasets where we can
obtain exactly same train/dev/test splits as the original DPR paper (NQ and TriviaQA), we tuned the
weight on the development set. For the remaining datasets, where splits are not available or the original DPR paper does not provide specific guidance, we tuned the weight on a subset of the training data. We obtained the optimal weight by performing grid search in the range [0, 2] with step size 0.05.

Hi @ccsasuke, my above comments have some typo regarding train/dev set.
Please this comment for details.

**Comment by ccsasuke** [comment]
Got it, thank you!
